### PR TITLE
Feature/more es6ify

### PR DESCRIPTION
**Pull request by eirslett**
A pretty big diff!
This is a bunch of cosmetic changes to the JS code. It's all ES2015 now! I've applied Airbnb's code style guide, with a few tweaks.
We run linting with `npm run lint`. @abesto Do you think it should be part of the gradle build? Set up with some kind of dependency?

### CODE REVIEW

File: zipkin-ui/test/page/common.test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import $ from 'jquery';
```
Comment (codefromthecrypt) [inline zipkin-ui/test/page/common.test.js:1]: whoah you can get money that easy?

**Comment by abesto** [comment]
There seems to be a missed dependency on `eslint-plugin-react`.

Here's what I'd do:

``` gradle
task npmLint(type: NpmTask) {
    args = ['run-script', 'lint']
    dependsOn npmInstall
}
check.dependsOn npmLint
```

Edit: this was implicit, but lemme state it explicitly: yes, I think making code quality checks part of the build process is great. I'm all for it.

File: zipkin-ui/test/page/common.test.js (hunk 1 above)
Comment (abesto) [inline zipkin-ui/test/page/common.test.js:1]: My browser says `import € from 'jquery'` is a syntax error :(

File: zipkin-ui/js/component_data/dependency.js [hunk 2]
```diff
@@ -45,37 +45,37 @@ export default component(function dependency() {
...
-      }.bind(this));
+      });
     });
 
-    this.on(document, 'parentChildDataRequested', function (event, {parent, child}) {
-      this.getDependencyData(parent, child, function (data) {
+    this.on(document, 'parentChildDataRequested', function(event, {parent, child}) {
+      this.getDependencyData(parent, child, data => {
         this.trigger(document, 'parentChildDataReceived', data);
-      }.bind(this));
```
Comment (eirslett) [inline zipkin-ui/js/component_data/dependency.js:62]: One of the arrow syntax perks: We don't have to worry about binding "this" all the time :smile:

**Comment by eirslett** [comment]
I rebased on top of master.
Could somebody please give me a +1 for merge?
(If there aren't any bugs, of course)

**Comment by codefromthecrypt** [comment]
ran the UI, clicked around and seems good to me.

:shipit:

**Comment by abesto** [comment]
:heart: :shipit:

**Pull request merged**